src/components/Inventory/index.tsx

```diff
@@ -146,11 +163,64 @@ export default function Inventory(props: InventoryProps): JSX.Element {
       facilityIds: filters.facilityIds,
       searchSortOrder,
     });
-    const updatedResult = apiSearchResults?.map((result) => {
-      const resultTyped = result as InventoryResult;
-      const facilityInventoriesNames = resultTyped.facilityInventories.map((nursery) => nursery.facility_name);
-      return { ...result, facilityInventories: facilityInventoriesNames.join('\r') };
-    });
+
+    let updatedResult: InventoryResultWithFacilityNames[] | undefined = [];
+    if (filters.facilityIds && filters.facilityIds.length) {
+      const nextResults: InventoryResultWithFacilityNames[] = [];
+      apiSearchResults?.forEach((result) => {
+        const resultTyped = result as FacilityInventoryResult;
+        const indexFound = nextResults.findIndex((res) => res.species_id === resultTyped.species_id);
+
+        if (indexFound !== undefined && indexFound !== -1) {
+          const existingSpecies = nextResults[indexFound];
+          nextResults[indexFound] = {
+            ...existingSpecies,
+            germinatingQuantity: (
+              Number(existingSpecies.germinatingQuantity) + Number(resultTyped['germinatingQuantity(raw)'])
+            ).toString(),
+            notReadyQuantity: (
+              Number(existingSpecies.notReadyQuantity) + Number(resultTyped['notReadyQuantity(raw)'])
+            ).toString(),
+            readyQuantity: (
+              Number(existingSpecies.readyQuantity) + Number(resultTyped['readyQuantity(raw)'])
+            ).toString(),
+            totalQuantity: (
+              Number(existingSpecies.totalQuantity) + Number(resultTyped['totalQuantity(raw)'])
+            ).toString(),
+            facilityInventories: `${existingSpecies.facilityInventories}, ${resultTyped.facility_name}`,
+          };
+        } else {
+          const transformedResult: InventoryResultWithFacilityNames = {
+            species_id: resultTyped.species_id,
+            species_scientificName: resultTyped.species_scientificName,
+            germinatingQuantity: resultTyped['germinatingQuantity(raw)'],
+            notReadyQuantity: resultTyped['notReadyQuantity(raw)'],
+            readyQuantity: resultTyped['readyQuantity(raw)'],
+            totalQuantity: resultTyped['totalQuantity(raw)'],
+            facilityInventories: resultTyped.facility_name,
+          };
+
+          nextResults.push(transformedResult);
+
+          // format results
+          updatedResult = nextResults.map((uR) => {
+            return {
+              ...uR,
+              germinatingQuantity: numericFormatter.format(uR.germinatingQuantity),
+              notReadyQuantity: numericFormatter.format(uR.notReadyQuantity),
+              readyQuantity: numericFormatter.format(uR.readyQuantity),
+              totalQuantity: numericFormatter.format(uR.totalQuantity),
```

Comment:
Thoughts on having the rendering component use the NumberFormatter component? Will simplify this file and keep semantics of number formatting in the right place. If this is to re-use existing component without further changes, this is fine as-is. Maybe use a reduce().map()

Reply:
Got it, leave it as-is 👍